### PR TITLE
Factorygirl to factorybot - fixes #1413

### LINKED ISSUE #1413: migrate from FactoryGirl to FactoryBot
because FactoryGirl is no more and long live FactoryBot.

### PR DESCRIPTION
**Pull request by logangingerich**
FactoryGirl fully switched over to FactoryBot with all tests passing.

### CODE REVIEW

**Comment by Br3nda** [comment]
`/home/travis/build/Growstuff/growstuff/Gemfile:127:26: C: Style/ExtraSpacing: Unnecessary spacing detected.`

**Comment by logangingerich** [comment]
I will definitely do this. Unfortunately, I'm somewhat unfamiliar with the process of squashing commits that have already been pushed. Any chance you could give me a quick rundown? Thanks!

**Comment by cesy** [comment]
Link to #1413

**Comment by pozorvlak** [comment]
@logangingerich Sure! It sounds like you already know how to squash commits locally? In which case

```
git checkout factorygirl-to-factorybot
git rebase --interactive dev
# squash the commits as usual
git push -f origin factorygirl-to-factorybot # assuming "origin" is your fork
```

The `-f` is needed because you're pushing a branch that doesn't include commits present in `origin` (namely, the ones that you just rewrote). It's often considered bad practice to force-push to a shared upstream repo (and we disallow it for our `dev` and `master` branches), but [it's OK to force-push to your own fork](https://blog.plover.com/prog/git-vacillation.html) because nobody has pulled from it.

**Comment by Br3nda** [comment]
@pozorvlak we can use the "Squash and merge" button on github when we merge this.
(they added that functionality earlier this year)

**Comment by Br3nda** [comment]
I dunno why hound didn't notice those. Hound is supposed to notice anything on a line that has changed.
I haven't checked but maybe those lines didn't change?

**Comment by pozorvlak** [comment]
@Br3nda aaah, that would make sense! Those lines didn't change, but the lines above them did, moving the first parameter of the hash and breaking alignment.

Re: squashing, I was thinking that the ideal history would be (commit 1 + commit 2) followed by commit 3, which we couldn't achieve with GitHub's "squash and merge". But I'm OK with this all being one commit.

File: spec/models/harvest_spec.rb [hunk 1]
```diff
@@ -47,110 +47,110 @@
...
   end
 
   context "stringification" do
-    let(:crop) { FactoryGirl.create(:crop, name: "apricot") }
+    let(:crop) { FactoryBot.create(:crop, name: "apricot") }
 
     it "apricots" do
-      @h = FactoryGirl.create(:harvest, crop: crop,
+      @h = FactoryBot.create(:harvest, crop: crop,
                                         quantity: nil,
```
Comment (pozorvlak) [inline spec/models/harvest_spec.rb]: This is the kind of thing that Rubocop's complaining about - `quantity`, `unit` and `weight_quantity` should be lined up with `crop`, so lines 154-156 should be dedented by one space. You should be able to fix these all automatically using [`rubocop -a`](https://github.com/bbatsov/rubocop/wiki/Automatic-Corrections).

[I hate this indentation style, for exactly this reason, but it's the [Rubocop default](https://rubocop.readthedocs.io/en/latest/cops_layout/#layoutalignhash) so we've gone with it to ease transition from other Rails projects.]

**Pull request merged**